script/apps/Aries/Creator/Game/blocks/block_types.lua

```diff
@@ -552,6 +552,24 @@ function block_types.LoadFromFile(filename)
 
 				attr.texture = LocalTextures:GetBlockTexture(attr.texture);
 
+				if (attr.randomTexture) then
+					local maxnum = 8;
+					local path, regex = attr.randomTexture:match("^(.*)/([^/\\]+)$");    
+					path = path or ""
+					regex = regex or attr.randomTexture;
+					local index = 2;
+					commonlib.Files.Find(nil, path, 0, 1000, function (f)
```

Comment:
I think there is no need to multiple textures.  Why not use a single big texture for all random blocks?

If you want to search you can find with wildcard like  `Files.Find("tex/abc*.*")`
But it may not be necessary if we use just one texture, like `abc_tile4x4`.  Use filename to store additional info.